FreeAPS/Sources/APS/OpenAPS/JavaScriptWorker.swift

```diff
@@ -20,18 +42,50 @@ final class JavaScriptWorker {
             }
         }
         let consoleLog: @convention(block) (String) -> Void = { message in
-            debug(.openAPS, "JavaScript log: \(message)")
+            let trimmedMessage = message.trimmingCharacters(in: .whitespacesAndNewlines)
+            if !trimmedMessage.isEmpty {
+//                debug(.openAPS, "JavaScript log: \(trimmedMessage)")
+                self.consoleLogs.append("\(trimmedMessage)")
+            }
         }
-
         context.setObject(
             consoleLog,
             forKeyedSubscript: "_consoleLog" as NSString
         )
         return context
     }
 
+    // New method to flush aggregated logs
+    private func outputLogs() {
+        var outputLogs = consoleLogs.joined(separator: "\n").trimmingCharacters(in: .whitespacesAndNewlines)
+        consoleLogs.removeAll()
+
+        if outputLogs.isEmpty { return }
+
```

Comment:
Example of log with no application of PR 345
```
2024-07-21T09:38:43-0700 [OpenAPS] OpenAPS.swift - determineBasal(currentTemp:clock:) - 21 - DEV: Start determineBasal
2024-07-21T09:38:45-0700 [OpenAPS] JavaScriptWorker.swift - createContext() - 23 - DEV: JavaScript log: Middleware reason: Nothing changed
2024-07-21T09:38:45-0700 [OpenAPS] JavaScriptWorker.swift - createContext() - 23 - DEV: JavaScript log: insulinFactor set to : 55
2024-07-21T09:38:45-0700 [OpenAPS] JavaScriptWorker.swift - createContext() - 23 - DEV: JavaScript log: Using weighted TDD average: 1 U, instead of past 24 h (0 U), weight: 0.65
```

Example of log with PR 345 applied but without the hack
```
2024-07-21T00:02:39-0700 [OpenAPS] OpenAPS.swift - determineBasal(currentTemp:clock:) - 21 - DEV: Start determineBasal
2024-07-21T00:02:41-0700 [OpenAPS] JavaScriptWorker.swift - outputLogs() - 78 - DEV: Middleware: Middleware reason: Nothing changed
2024-07-21T00:02:41-0700 [OpenAPS] JavaScriptWorker.swift - outputLogs() - 78 - DEV: Middleware: insulinFactor set to : 55
2024-07-21T00:02:41-0700 [OpenAPS] JavaScriptWorker.swift - outputLogs() - 78 - DEV: Middleware: Using weighted TDD average: 1 U, instead of past 24 h (0 U), weight: 0.65
2024-07-21T00:02:41-0700 [OpenAPS] JavaScriptWorker.swift - outputLogs() - 78 - DEV: Middleware: , Dynamic ratios log: Dynamic Settings disabled
```
and finally example of the logs post hack
```
2024-07-21T12:14:31-0700 [OpenAPS] OpenAPS.swift - determineBasal(currentTemp:clock:) - 21 - DEV: Start determineBasal
2024-07-21T12:14:33-0700 [OpenAPS] JavaScriptWorker.swift - createContext() - 49 - DEV: JavaScript log: Middleware reason: Nothing changed
2024-07-21T12:14:33-0700 [OpenAPS] JavaScriptWorker.swift - createContext() - 49 - DEV: JavaScript log: insulinFactor set to : 55
2024-07-21T12:14:33-0700 [OpenAPS] JavaScriptWorker.swift - createContext() - 49 - DEV: JavaScript log: Using weighted TDD average: 1 U, instead of past 24 h (0 U), weight: 0.65
2024-07-21T12:14:33-0700 [OpenAPS] JavaScriptWorker.swift - createContext() - 49 - DEV: JavaScript log: , Dynamic ratios log: Dynamic Settings disabled
2
```

Final action was to restore code to dev commit 8303f0a8 with no patches applied to get original logs again.
These started at 12:46 local. (Search for `Trio Started` to locate the beginning of each rebuild.) I let this run for 3 full cycles before issuing the log one more time.

[2024-07-21-MDT-pr-345-on-off-hack-off-log.txt](https://github.com/user-attachments/files/16325914/2024-07-21-MDT-pr-345-on-off-hack-off-log.txt)

Reply:
Not exactly sure I agreed with the change. 😅

The log shows what script was ran to output it. If the wrong script is being showed, we should fix the calling method.

And a hack means we won’t address this issue properly in the future.

Can we merge this PR as is, and create an issue for the Middleware script name? 

Agile development. Baby steps. 😅

Reply:
Will see what I can do 💪

Reply:
possible solution was submitted